system/BaseModel.php

```diff
@@ -1636,11 +1637,13 @@ public function __isset(string $name): bool
      */
     public function __call(string $name, array $params)
     {
-        if (method_exists($this->db, $name)) {
-            return $this->db->{$name}(...$params);
+        if (! method_exists($this->db, $name)) {
+            $className = static::class;
+
+            throw new BadMethodCallException('Call to undefined method ' . $className . '::' . $name);
```

Comment:
It seems this is a breaking change.

Reply:
Yes, this is breaking for users using the `BaseModel` for their NoSQL models.

Reply:
@kenjis @paulbalandan 

Yes, It's a breaking change, that's for sure. However, I don't understand what the problem is.
I don't know what use case it would be better to call a non-existent method and return null when using NoSQL.
What do NoSQL users call non-existent methods for?

Reply:
We just can't merge breaking changes until the next major version (v5.0).
Because if we change the public API, user's applications may break.

Reply:
The problem is we do not know how they implemented this in their code. One I can think of is like this:
```php
class NoSqlModel extends BaseModel
{
    ...
}

// then somewhere in their code
$nosql = new NoSqlModel();

if ($nosql->doSomething() === null) {
    // do something else
}
```

In code above, the user expects a call to non-existent method to return `null`. However, if this be merged, their code would break because it will now throw `BadMethodCallException` instead.

Reply:
@kenjis @paulbalandan 

> We just can't merge breaking changes until the next major version (v5.0).
> The problem is we do not know how they implemented this in their code.

I understand.
Thank you.
FIxed it.